lib/engines/content_block_manager/test/components/content_block/edition/details/fields/array/array_item_component_test.rb

```diff
@@ -32,30 +34,52 @@ class ContentBlockManager::ContentBlockEdition::Details::Fields::Array::ItemComp
     let(:field_value) { ["", "Some text"] }
     let(:index) { 1 }
 
-    it "renders a text field" do
-      render_inline(component)
+    describe "if the item can be deleted" do
+      let(:can_be_deleted) { true }
 
-      assert_selector "label", text: "Bar"
-      assert_selector "input[type='text'][value='Some text'][name='foo[bar][]'][id='foo_bar_1']"
-    end
+      it "renders with the correct class" do
+        render_inline(component)
 
-    describe "when error messages are present" do
-      let(:errors_for_field) do
-        [{ text: "Bar cannot be blank" }]
+        assert_selector ".app-c-content-block-manager-array-item-component"
+        refute_selector ".app-c-content-block-manager-array-item-component--immutable"
       end
 
-      before do
-        component.expects(:errors_for).with(errors, "#{error_lookup_prefix}_#{index}".to_sym).returns(errors_for_field)
+      it "renders a text field" do
+        render_inline(component)
+
+        assert_selector "label", text: "Bar"
+        assert_selector "input[type='text'][value='Some text'][name='foo[bar][]'][id='foo_bar_1']"
       end
 
-      it "renders an error" do
-        render_inline(component)
+      describe "when error messages are present" do
+        let(:errors_for_field) do
+          [{ text: "Bar cannot be blank" }]
+        end
 
-        assert_selector ".govuk-form-group--error" do |form_group|
-          form_group.assert_selector ".govuk-error-message", text: "Bar cannot be blank"
+        before do
+          component.expects(:errors_for).with(errors, "#{error_lookup_prefix}_#{index}".to_sym).returns(errors_for_field)
+        end
+
+        it "renders an error" do
+          render_inline(component)
+
+          assert_selector ".govuk-form-group--error" do |form_group|
+            form_group.assert_selector ".govuk-error-message", text: "Bar cannot be blank"
+          end
         end
       end
     end
+
+    describe "if the item cannot be deleted" do
+      let(:can_be_deleted) { false }
+
+      it "renders with the correct class" do
```

Comment:
again, perhaps I'd find it easier if we spell out explicitly which class we expect and why, e.g

```
it "renders without the '--immutable' class, to suppress the 'delete' button" do
```